pkg/server/container_stats_list_unix.go

```diff
@@ -61,19 +62,38 @@ func (c *criService) containerMetrics(
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to extract container metrics")
 		}
-		metrics := s.(*v1.Metrics)
```

Comment:
@Random-Liu - This would have previously panic'd if this wasnt the return type. Should we have a `default` case where we return error or maintain this?

Reply:
The CRI expects failure on conversion errors. We should not panic... we should return the error.